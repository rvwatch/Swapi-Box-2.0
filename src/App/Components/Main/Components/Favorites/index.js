import React, { Component } from 'react';
import { array, func } from 'prop-types';
import './Favorites.css';
import Card from '../Card';

class Favorites extends Component {
  render() {
    const createCards = !this.props.favorites.length ? (
      <h1 className="no-favs-message">No Favorites Selected!</h1>
    ) : (
      this.props.favorites.map((favorite, index) => (
        <Card
          toggleFavorites={this.props.toggleFavorites}
          cardData={favorite}
          isFav={true}
          {...favorite}
          key={favorite.name + index}
        />
      ))
    );

    return <section>{createCards}</section>;
  }
}

Favorites.propTypes = {
  favorites: array,
  toggleFavorites: func
};

export default Favorites;
