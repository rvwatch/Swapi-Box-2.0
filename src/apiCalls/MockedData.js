export const mockUrl = 'starwars.com';

export const mockUrlArray = ['starwars.com', 'starwars.com', 'starwars.com'];

export const mockSingleFilm = {
  title: 'Movie!',
  opening_crawl: 'Movie text!'
};

export const mockApiFilm = {
  results: [
    { title: 'Movie!', opening_crawl: 'Movie text!' },
    { title: 'Movie!', opening_crawl: 'Movie text!' },
    { title: 'Movie!', opening_crawl: 'Movie text!' }
  ]
};

export const mockPerson = {
  name: 'Luke Skywalker',
  homeworld: 'Tatooine',
  population: 200000,
  species: 'Human'
};

export const mockApiPeople = {
  results: [
    {
      name: 'Luke Skywalker',
      homeworld: 'Tatooine',
      population: 200000,
      species: 'Human'
    },
    {
      name: 'C-3PO',
      homeworld: 'Tatooine',
      population: 200000,
      species: 'Droid'
    },
    {
      name: 'Obi-Wan Kenobi',
      homeworld: 'Stewjon',
      population: 'unknown',
      species: 'Human'
    }
  ]
};

export const mockCleanPeople = [
  {
    homeworld: 'Homeworld: undefined',
    id: 'Luke Skywalker0',
    name: 'Name: Luke Skywalker',
    population: 'Population: undefined',
    species: 'Species: undefined'
  },
  {
    homeworld: 'Homeworld: undefined',
    id: 'C-3PO1',
    name: 'Name: C-3PO',
    population: 'Population: undefined',
    species: 'Species: undefined'
  },
  {
    homeworld: 'Homeworld: undefined',
    id: 'Obi-Wan Kenobi2',
    name: 'Name: Obi-Wan Kenobi',
    population: 'Population: undefined',
    species: 'Species: undefined'
  }
];

export const mockPlanetData = {
  results: [
    {
      name: 'Tatooine',
      rotation_period: '23',
      orbital_period: '304',
      diameter: '10465',
      climate: 'arid',
      gravity: '1 standard',
      terrain: 'desert',
      surface_water: '1',
      population: '200000',
      residents: ['']
    },
    {
      name: 'Tatooine',
      rotation_period: '23',
      orbital_period: '304',
      diameter: '10465',
      climate: 'arid',
      gravity: '1 standard',
      terrain: 'desert',
      surface_water: '1',
      population: '200000',
      residents: ['']
    },
    {
      name: 'Tatooine',
      rotation_period: '23',
      orbital_period: '304',
      diameter: '10465',
      climate: 'arid',
      gravity: '1 standard',
      terrain: 'desert',
      surface_water: '1',
      population: '200000',
      residents: ['']
    }
  ]
};

export const mockCleanPlanets = [
  {
    climate: 'Climate: arid',
    id: 'Tatooine0',
    name: 'Name: Tatooine',
    population: 'Population: 200000',
    residents: [{ name: undefined }],
    terrain: 'Terrain: desert'
  },
  {
    climate: 'Climate: arid',
    id: 'Tatooine1',
    name: 'Name: Tatooine',
    population: 'Population: 200000',
    residents: [{ name: undefined }],
    terrain: 'Terrain: desert'
  },
  {
    climate: 'Climate: arid',
    id: 'Tatooine2',
    name: 'Name: Tatooine',
    population: 'Population: 200000',
    residents: [{ name: undefined }],
    terrain: 'Terrain: desert'
  }
];

export const mockResidents = [
  { name: 'Luke Skywalker' },
  { name: 'Luke Skywalker' },
  { name: 'Luke Skywalker' }
];

export const mockVehicleData = {
  results: [
    {
      name: 'Sand Crawler',
      model: 'Digger Crawler',
      manufacturer: 'Corellia Mining Corporation',
      cost_in_credits: '150000',
      length: '36.8',
      max_atmosphering_speed: '30',
      crew: '46',
      passengers: '30',
      cargo_capacity: '50000',
      consumables: '2 months',
      vehicle_class: 'wheeled',
      pilots: [],
      films: ['https://swapi.co/api/films/5/', 'https://swapi.co/api/films/1/'],
      created: '2014-12-10T15:36:25.724000Z',
      edited: '2014-12-22T18:21:15.523587Z',
      url: 'https://swapi.co/api/vehicles/4/'
    }
  ]
};

export const mockCleanVehicle = [
  {
    id: 'Sand Crawler0',
    model: 'Model: Digger Crawler',
    name: 'Name: Sand Crawler',
    passengers: 'Passengers: 30',
    vehicle_class: 'Class: wheeled'
  }
];

export const mockSelectedCard1 = {
  name: 'Luke Skywalker',
  homeworld: 'Tatooine',
  population: 200000,
  species: 'Human',
  id: 1
};

export const mockSelectedCard2 = {
  name: 'Luke Skywalker',
  homeworld: 'Tatooine',
  population: 200000,
  species: 'Human',
  id: 2
};

export const mockSelectedCard3 = {
  name: 'Luke Skywalker',
  homeworld: 'Tatooine',
  population: 200000,
  species: 'Human',
  id: 3
};

export const mockFavorites = [
  {
    name: 'Luke Skywalker',
    homeworld: 'Tatooine',
    population: 200000,
    species: 'Human',
    id: 1
  },
  {
    name: 'Luke Skywalker',
    homeworld: 'Tatooine',
    population: 200000,
    species: 'Human',
    id: 2
  },
  {
    name: 'Luke Skywalker',
    homeworld: 'Tatooine',
    population: 200000,
    species: 'Human',
    id: 3
  }
];

export const mockStarshipData = {
  results: [
    {
      MGLT: '40',
      cargo_capacity: '250000000',
      consumables: '6 years',
      cost_in_credits: '1143350000',
      created: '2014-12-15T12:31:42.547000Z',
      crew: '279144',
      edited: '2017-04-19T10:56:06.685592Z',
      films: (2)[
        ('https://swapi.co/api/films/2/', 'https://swapi.co/api/films/3/')
      ],
      hyperdrive_rating: '2.0',
      length: '19000',
      manufacturer: 'Kuat Drive Yards, Fondor Shipyards',
      max_atmosphering_speed: 'n/a',
      model: 'Executor-class star dreadnought',
      name: 'Executor',
      passengers: '38000',
      pilots: [],
      starship_class: 'Star dreadnought',
      url: 'https://swapi.co/api/starships/15/'
    }
  ]
};

export const mockCleanStarship = [
  {
    crew: 'Crew: 279144',
    id: 'Executor0',
    length: 'Length: 19000m/s',
    model: 'Model: Executor-class star dreadnought',
    name: 'Name: Executor',
    passengers: 'Passengers: 38000',
    starship_class: 'Class: Star dreadnought'
  }
];
