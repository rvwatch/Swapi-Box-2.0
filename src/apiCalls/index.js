export const getData = async url => {
  try {
    const response = await fetch(url);
    const starWarsData = await response.json();
    return starWarsData;
  } catch (errs) {
    throw new Error('getData has thrown an error');
  }
};

export const getScrollText = async url => {
  const filmData = await getData(url);
  const selectFilm = Math.floor(Math.random() * filmData.results.length);
  const { title, opening_crawl } = filmData.results[selectFilm];
  return { title, opening_crawl };
};

export const getPeopleData = async url => {
  const peopleData = await getData(url);

  const peoplePromises = peopleData.results.map(async (person, index) => {
    const homeworld = await getData(person.homeworld);
    const species = await getData(person.species);
    return {
      name: person.name,
      homeworld: homeworld.name,
      population: homeworld.population,
      species: species.name,
      id: person.name + index
    };
  });
  return Promise.all(peoplePromises);
};

export const getPlanetData = async url => {
  const planetData = await getData(url);

  const planetPromises = planetData.results.map(async (planet, index) => {
    const { name, terrain, population, climate, residents } = planet;
    const residentList = await getResidents(residents);
    return {
      name,
      terrain,
      population,
      climate,
      residents: [...residentList],
      id: name + index
    };
  });
  return Promise.all(planetPromises);
};

export const getResidents = async residentUrls => {
  const residentPromises = residentUrls.map(async url => {
    const person = await getData(url);
    return { name: person.name };
  });
  return Promise.all(residentPromises);
};

export const getVehicleData = async url => {
  const vehicleData = await getData(url);

  const vehiclePromises = vehicleData.results.map(async (vehicle, index) => {
    const { name, model, vehicle_class, passengers } = vehicle;
    return {
      name,
      model,
      vehicle_class,
      passengers,
      id: name + index
    };
  });
  return Promise.all(vehiclePromises);
};

export const getStarshipData = async url => {
  const starshipData = await getData(url);

  const starshipPromises = starshipData.results.map(async (ship, index) => {
    const { name, model, starship_class, passengers, crew, length } = ship;
    return {
      name,
      model,
      starship_class,
      passengers,
      crew,
      length,
      id: name + index
    };
  });
  return Promise.all(starshipPromises);
};
