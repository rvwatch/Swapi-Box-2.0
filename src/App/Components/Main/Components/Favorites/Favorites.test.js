import React from 'react';
import { shallow } from 'enzyme';
import Favorites from './index';
import { mockFavorites } from '../../../../../apiCalls/MockedData';

describe('Favorites ', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(<Favorites favorites={mockFavorites} />);
  });
  it('renders correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
