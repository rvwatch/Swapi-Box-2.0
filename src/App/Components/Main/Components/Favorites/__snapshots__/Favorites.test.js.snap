// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Favorites  renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Favorites
    favorites={
        Array [
            Object {
              "homeworld": "Tatooine",
              "id": 1,
              "name": "Luke Skywalker",
              "population": 200000,
              "species": "Human",
            },
            Object {
              "homeworld": "Tatooine",
              "id": 2,
              "name": "Luke Skywalker",
              "population": 200000,
              "species": "Human",
            },
            Object {
              "homeworld": "Tatooine",
              "id": 3,
              "name": "Luke Skywalker",
              "population": 200000,
              "species": "Human",
            },
          ]
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Card
          cardData={
                    Object {
                              "homeworld": "Tatooine",
                              "id": 1,
                              "name": "Luke Skywalker",
                              "population": 200000,
                              "species": "Human",
                            }
          }
          homeworld="Tatooine"
          id={1}
          isFav={true}
          name="Luke Skywalker"
          population={200000}
          species="Human"
          toggleFavorites={undefined}
/>,
        <Card
          cardData={
                    Object {
                              "homeworld": "Tatooine",
                              "id": 2,
                              "name": "Luke Skywalker",
                              "population": 200000,
                              "species": "Human",
                            }
          }
          homeworld="Tatooine"
          id={2}
          isFav={true}
          name="Luke Skywalker"
          population={200000}
          species="Human"
          toggleFavorites={undefined}
/>,
        <Card
          cardData={
                    Object {
                              "homeworld": "Tatooine",
                              "id": 3,
                              "name": "Luke Skywalker",
                              "population": 200000,
                              "species": "Human",
                            }
          }
          homeworld="Tatooine"
          id={3}
          isFav={true}
          name="Luke Skywalker"
          population={200000}
          species="Human"
          toggleFavorites={undefined}
/>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "Luke Skywalker0",
        "nodeType": "function",
        "props": Object {
          "cardData": Object {
            "homeworld": "Tatooine",
            "id": 1,
            "name": "Luke Skywalker",
            "population": 200000,
            "species": "Human",
          },
          "homeworld": "Tatooine",
          "id": 1,
          "isFav": true,
          "name": "Luke Skywalker",
          "population": 200000,
          "species": "Human",
          "toggleFavorites": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "Luke Skywalker1",
        "nodeType": "function",
        "props": Object {
          "cardData": Object {
            "homeworld": "Tatooine",
            "id": 2,
            "name": "Luke Skywalker",
            "population": 200000,
            "species": "Human",
          },
          "homeworld": "Tatooine",
          "id": 2,
          "isFav": true,
          "name": "Luke Skywalker",
          "population": 200000,
          "species": "Human",
          "toggleFavorites": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "Luke Skywalker2",
        "nodeType": "function",
        "props": Object {
          "cardData": Object {
            "homeworld": "Tatooine",
            "id": 3,
            "name": "Luke Skywalker",
            "population": 200000,
            "species": "Human",
          },
          "homeworld": "Tatooine",
          "id": 3,
          "isFav": true,
          "name": "Luke Skywalker",
          "population": 200000,
          "species": "Human",
          "toggleFavorites": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "section",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Card
            cardData={
                        Object {
                                    "homeworld": "Tatooine",
                                    "id": 1,
                                    "name": "Luke Skywalker",
                                    "population": 200000,
                                    "species": "Human",
                                  }
            }
            homeworld="Tatooine"
            id={1}
            isFav={true}
            name="Luke Skywalker"
            population={200000}
            species="Human"
            toggleFavorites={undefined}
/>,
          <Card
            cardData={
                        Object {
                                    "homeworld": "Tatooine",
                                    "id": 2,
                                    "name": "Luke Skywalker",
                                    "population": 200000,
                                    "species": "Human",
                                  }
            }
            homeworld="Tatooine"
            id={2}
            isFav={true}
            name="Luke Skywalker"
            population={200000}
            species="Human"
            toggleFavorites={undefined}
/>,
          <Card
            cardData={
                        Object {
                                    "homeworld": "Tatooine",
                                    "id": 3,
                                    "name": "Luke Skywalker",
                                    "population": 200000,
                                    "species": "Human",
                                  }
            }
            homeworld="Tatooine"
            id={3}
            isFav={true}
            name="Luke Skywalker"
            population={200000}
            species="Human"
            toggleFavorites={undefined}
/>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "Luke Skywalker0",
          "nodeType": "function",
          "props": Object {
            "cardData": Object {
              "homeworld": "Tatooine",
              "id": 1,
              "name": "Luke Skywalker",
              "population": 200000,
              "species": "Human",
            },
            "homeworld": "Tatooine",
            "id": 1,
            "isFav": true,
            "name": "Luke Skywalker",
            "population": 200000,
            "species": "Human",
            "toggleFavorites": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "Luke Skywalker1",
          "nodeType": "function",
          "props": Object {
            "cardData": Object {
              "homeworld": "Tatooine",
              "id": 2,
              "name": "Luke Skywalker",
              "population": 200000,
              "species": "Human",
            },
            "homeworld": "Tatooine",
            "id": 2,
            "isFav": true,
            "name": "Luke Skywalker",
            "population": 200000,
            "species": "Human",
            "toggleFavorites": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "Luke Skywalker2",
          "nodeType": "function",
          "props": Object {
            "cardData": Object {
              "homeworld": "Tatooine",
              "id": 3,
              "name": "Luke Skywalker",
              "population": 200000,
              "species": "Human",
            },
            "homeworld": "Tatooine",
            "id": 3,
            "isFav": true,
            "name": "Luke Skywalker",
            "population": 200000,
            "species": "Human",
            "toggleFavorites": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "section",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
    "disableLifecycleMethods": true,
  },
}
`;
